mytoyota/models/vehicle.py

```diff
@@ -264,8 +264,8 @@ async def get_summary(
         resp = await self._api.get_trips_endpoint(
             self.vin, from_date, to_date, summary=True, limit=1, offset=0
         )
-        if resp.payload is None:
-            return None
+        if resp.payload is None or len(resp.payload.summary) == 0:
+            return []
```

Comment:
Why did you decide to return an empty list instead of `None`?
That no longer matches the specified return value of the function, does it?
Instead of `Optional[List[Summary]]`, we now actually have `List[Optional[Summary]]`.

Reply:
This means that the user does not need to check for None before looping on the result. It makes no difference to the output of the application and saves the check. 

The only reason (I think) to change it back would be if we want to differentiate between cars that don't support summaries(return None) and when no data was returned(return []). So far no one has reported a car which returned None(the code would have thrown an exception in simple_client_example.py)

Reply:
That works for me. We should just adapt the type hint accordingly.

Reply:
Updated.